### PR TITLE
Change application reserved error code range

### PR DESCRIPTION
**Pull request by lexs**
We now have codes up to 204, so application needs to start at 205.

### CODE REVIEW

**Comment by lexs** [comment]
Should we bump this number to 300 to allow us to add more error codes?

File: Protocol.md [hunk 1]
```diff
@@ -349,7 +349,7 @@ The Error Data is typically an Exception message, but could include stringified
 | __RESERVED__                   | 0xFFFFFFFF | __Reserved for Extension Use__ |
 
 __NOTE__: Values in the range of 0x0001 to 0x00FF are reserved for use as SETUP error codes. Values in the range of
-0x00101 to 0x001FF are reserved for connection error codes. Values in the range of 0x00201 to 0xFFFFFFFE are reserved for application layer
+0x00101 to 0x001FF are reserved for connection error codes. Values in the range of 0x00205 to 0xFFFFFFFE are reserved for application layer
```
Comment (yschimke) [inline Protocol.md]: I think the intent was that errors where in distinct blocks, somewhat like 2XX, 3XX, 4XX, 5XX in HTTP.

So your suggestion to go to 0x300 seems like the only sensible approach.

Comment (benjchristensen) [inline Protocol.md]: 0x301 seems good to me.

**Comment by benjchristensen** [comment]
@tmontgomery what do you recommend?

File: Protocol.md (hunk 1 above)
Comment (tmontgomery) [inline Protocol.md]: I agree. 0x301.

**Comment by tmontgomery** [comment]
Saving some space is good. 0x301 should be OK.

**Comment by lexs** [comment]
Updated to 0x301.

File: Protocol.md [hunk 2]
```diff
@@ -349,7 +349,7 @@ The Error Data is typically an Exception message, but could include stringified
 | __RESERVED__                   | 0xFFFFFFFF | __Reserved for Extension Use__ |
 
 __NOTE__: Values in the range of 0x0001 to 0x00FF are reserved for use as SETUP error codes. Values in the range of
-0x00101 to 0x001FF are reserved for connection error codes. Values in the range of 0x00201 to 0xFFFFFFFE are reserved for application layer
+0x00101 to 0x001FF are reserved for connection error codes. Values in the range of 0x00301 to 0xFFFFFFFE are reserved for application layer
```
Comment (yschimke) [inline Protocol.md]: This text currently leaves a gap e.g. doesn't describe 0x00201 to 0x002FF

Comment (tmontgomery) [inline Protocol.md]: Good catch. 0x201 to 0x301 should be marked for later use in future versions.

Comment (yschimke) [inline Protocol.md]: You've confused me more!  But it's late here.

Comment (tmontgomery) [inline Protocol.md]: Sorry. Meant that those (if I recall correctly) are for space for use by later versions of the protocol. "Reserved for future use".

**Comment by benjchristensen** [comment]
Is this ready for merging?

**Comment by yschimke** [comment]
No, I think it should be changed to explain the 4 ranges  
0x0001 to 0x00FF - setup errors
0x0101 to 0x01FF - connection errors
0x0201 to 0x02FF - predefined application errors
0x0301 to 0xFFFFFFFE? - custom application errors

starting custom application errors at 0x0205 seems like it needlessly precludes the spec from ever defining more.

**Comment by lexs** [comment]
Updated the NOTE wording, should be ready now.

**Pull request merged**